debezium-server/debezium-server-nats_streaming/src/main/java/io/debezium/server/nats_streaming/NatsStreamingChangeConsumer.java

```diff
@@ -0,0 +1,141 @@
+    private String clientID;
+
+    @ConfigProperty(name = PROP_PREFIX + "null.key", defaultValue = "default")
+    String nullKey;
+
+    private StreamingConnection sc;
+
+    @Inject
+    @CustomConsumerBuilder
+    Instance<StreamingConnection> customStreamingConnection;
+
+    @PostConstruct
+    void connect() {
+        if (customStreamingConnection.isResolvable()) {
+            sc = customStreamingConnection.get();
+            LOGGER.info("Obtained custom configured StreamingConnection '{}'", sc);
+            return;
+        }
+
+        // Read config
+        final Config config = ConfigProvider.getConfig();
+        url = config.getValue(PROP_URL, String.class);
+        clusterID = config.getValue(PROP_CLUSTER_ID, String.class);
+        clientID = config.getValue(PROP_CLIENT_ID, String.class);
+
+        // Setup NATS Streaming connection
+        Options stanOptions = new Options.Builder()
+                .natsUrl(url)
+                .connectionListener(new ConnectionListener() {
+                    public void connectionEvent(Connection natsConnection, Events event) {
+                        if (event == Events.DISCONNECTED) {
+                            Panic("NATS Streaming disconnected.");
+                        }
+                    }
+                })
+                .connectionLostHandler((streamingConnection, e) -> {
+                    Panic("NATS Streaming connection lost.");
+                })
+                .build();
+        try {
+            sc = NatsStreaming.connect(clusterID, clientID, stanOptions);
+        }
+        catch (Exception e) {
+            throw new DebeziumException(e);
+        }
+
+        LOGGER.info("Using default StreamingConnection '{}'", sc);
+    }
+
+    @PreDestroy
+    void close() {
+        try {
+            sc.close();
+            LOGGER.info("NATS Streaming connection closed.");
+        }
+        catch (Exception e) {
+            // Do nothing
+        }
+    }
+
+    @Override
+    public void handleBatch(List<ChangeEvent<Object, Object>> records,
+                            RecordCommitter<ChangeEvent<Object, Object>> committer)
+            throws InterruptedException {
+
+        for (ChangeEvent<Object, Object> record : records) {
+            if (record.value() != null) {
+                String subject = streamNameMapper.map(record.destination());
+                byte[] recordBytes = getBytes(record.value());
+                LOGGER.trace("Received event @ {} = '{}'", subject, getString(record.value()));
+
+                try {
+                    sc.publish(subject, recordBytes);
+                }
+                catch (Exception e) {
+                    throw new DebeziumException(e);
+                }
+            }
+            committer.markProcessed(record);
+        }
+        committer.markBatchFinished();
+    }
+
+    public static void Panic(String message) {
+        LOGGER.error(message);
+        Runtime.getRuntime().halt(0);
```

Comment:
I don't think the error should be handled this way. I'd prefer an exception thrown etc.

Reply:
Now throwing a DebeziumException, done!

Reply:
Now throwing a `DebeziumException`, done!